Automattic/woocommerce-payments
# Add filter to disable WooPay checkout auto-redirect and email OTP iframe

Fixes 2298-gh-Automattic/woopay

#### Changes proposed in this Pull Request

- Add filter to disable the WooPay email input hooks.

#### Testing instructions

1. Switch to this branch.
2. Run `npm run build:client`.
3. Install [Code Snippets](https://wordpress.org/plugins/code-snippets/) and enable the following filter:
```php
add_filter( 'wcpay_is_woopay_email_input_enabled', '__return_false' );
```
4. Ensure the checkout email input doesn't trigger the OTP iframe.
5. Ensure the checkout email input doesn't trigger auto-redirection to WooPay if the merchant site is under HTTPS and user already logged-in to WooPay.
6. Ensure the email input hooks work as expected if the filter is disabled.

Please test steps 4 - 6 with blocks and classic checkout.

-------------------


**Post merge**
